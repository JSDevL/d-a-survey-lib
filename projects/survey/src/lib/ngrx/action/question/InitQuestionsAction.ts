import {QuestionAction} from './QuestionAction';
import {Question} from '../../../model/question/Question';

export class InitQuestionsAction extends QuestionAction {
  public readonly type = 'INIT_QUESTION_ACTION';
  private readonly payload: Question[];

  constructor(payload: Question[]) {
    super();
    this.payload = payload;
  }

  apply(state: Question[]): Question[] {
    if (this.payload) {
      return [...this.payload];
    }

    return state;
  }
}
