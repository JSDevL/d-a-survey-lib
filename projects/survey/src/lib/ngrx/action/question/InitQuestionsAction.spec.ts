import {InitQuestionsAction} from './InitQuestionsAction';

describe('InitQuestionsAction', () => {
  it('should not affect state when apply(falsy or !Question[])', () => {
    const action = new InitQuestionsAction(null);
    const state = [];

    expect(action.apply(state)).toEqual(state);
  });

  it('should replace state with payload when apply(Question[])', () => {
    const payload = [
      null,
      null
    ];
    const action = new InitQuestionsAction(payload);

    expect(action.apply([])).toEqual(payload);
  });
});
