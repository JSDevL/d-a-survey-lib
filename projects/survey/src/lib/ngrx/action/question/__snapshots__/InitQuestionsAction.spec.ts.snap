// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitQuestionsAction should not affect state apply(falsy or !Question[]) 1`] = `undefined`;

exports[`InitQuestionsAction should replace state if apply(Question[]) 1`] = `undefined`;
